featurebyte/query_graph/sql/feature_historical.py

```diff
@@ -361,87 +427,27 @@ def get_historical_features_query_set(
     -------
     FeatureQuerySet
     """
-    # Process nodes in batches
-    node_groups = split_nodes(graph, nodes, NUM_FEATURES_PER_QUERY, source_info)
-
-    if len(node_groups) == 1:
-        # Fallback to simpler non-batched query if there is only one group to avoid overhead
-        sql_expr, _ = get_historical_features_expr(
```

Comment:
Another changes is that this logic of falling back to non-bached query is removed. The reason is that the query might fail even if the initial batch size determines that there is only one batch.